R/target_market_share.R

```diff
@@ -316,15 +315,72 @@ target_market_share <- function(data,
   data <- data %>%
     pivot_wider2()
 
+  data <- data %>%
+    left_join(green_or_brown, by = c("sector", "technology")) %>%
+    left_join(tmsr_or_smsp, by = "green_or_brown") %>%
+    rename(target_name = .data$which_metric) %>%
+    select(-.data$green_or_brown)
+
   corporate_economy <- calculate_ald_benchmark(ald, region_isos, by_company)
 
   relevant_sectors <- unique(data$sector)
 
   relevant_corporate_economy <- corporate_economy %>%
     filter(.data$sector %in% relevant_sectors)
 
+  relevant_corporate_economy <- relevant_corporate_economy %>%
+    left_join(green_or_brown, by = c("sector", "technology")) %>%
+    left_join(tmsr_or_smsp, by = "green_or_brown") %>%
+    rename(target_name = .data$which_metric) %>%
+    select(-.data$green_or_brown)
+
+  data <- data %>%
+    dplyr::bind_rows(relevant_corporate_economy)
+
+  percent_by_sector_groups <- maybe_add_name_ald(
+    c("sector", "region", "scenario_source", "metric"),
+    by_company
+  )
+
+  data <- data %>%
+    group_by(!!!rlang::syms(c(percent_by_sector_groups, "technology"))) %>%
+    arrange(.data$year) %>%
+    mutate(initial_technology_production = first(.data$production)) %>%
+    group_by(!!!rlang::syms(c(percent_by_sector_groups, "year"))) %>%
+    mutate(
+      .sector_production = sum(.data$production)
+      ) %>%
+    group_by(!!!rlang::syms(percent_by_sector_groups)) %>%
+    mutate(
+      initial_sector_production = first(.data$.sector_production),
+      .sector_production = NULL
```

Comment:
generally I prefer to use the `select(-{varname})` syntax for removing variables (as you've done elsewhere in this PR) because I feel it is more obvious/explicit what it does

Reply:
Hmm, ya I was waffling between these two approaches actually. I do like the `.varname = blah, usefulvar = function(.varname), .varname = NULL`, when you are just creating an interim column that only gets used in one mutate call, but I think that only makes sense if it can all fit into the same mutate call. 

I think I agree that the `select()` approach works better here.